app/middlewares/user.js

```diff
@@ -32,3 +34,51 @@ exports.verifyPassword = (req, _, next) => {
   if (verifyPassword(password)) return next();
   return next(errors.badRequestError("password doesn't meet the required characteristics"));
 };
+
+exports.userSchema = () => {
+  const check = checkSchema({
+    firstName: {
+      exists: {
+        errorMessage: 'First name should exist'
+      },
+      isString: {
+        errorMessage: 'First name should be string'
+      }
+    },
+    lastName: {
+      exists: {
+        errorMessage: 'Last name should exist'
+      },
+      isString: {
+        errorMessage: 'Last name should be string'
+      }
+    },
+    email: {
+      exists: {
+        errorMessage: 'Email should exist'
+      },
+      isString: {
+        errorMessage: 'Email should be string'
+      }
+    },
+    password: {
+      exists: {
+        errorMessage: 'Password should exist'
+      },
+      isString: {
+        errorMessage: 'Password name should be string'
+      }
+    }
+  });
+  return check;
+};
+
+exports.userSchemaValidator = (req, _, next) => {
+  const errorsValidation = validationResult(req);
+  if (!errorsValidation.isEmpty()) {
+    const errorReq = errorsValidation.array({ onlyFirstError: true });
+    const error = first(errorReq);
+    return next(errors.badRequestError(error ? error.msg : 'Bad request'));
+  }
+  return next();
+};
```

Comment:
Like this, the validator only returns the first error, if the body is missing two properties it will only give us notice of a single error, what you can do is use the `mapped` function that has `errorsValidation` available and this simply pass it to the custom error `badRequestError`.

Reply:
This validator would not only be for users, so it can only be called SchemaValidator and be in a separate file.